### ISSUE TITLE
`1.20.4` -> `1.20.5-1.20.6`: Items with incorrectly formatted tags cause an immediate disconnect (NBT ban)

### ISSUE THREAD

**Issue by sodiboo** [ViaVersion/ViaVersion#3935]
### `/viaversion dump` Output

https://dump.viaversion.com/9cec12825e49b9245f348c55163d04f2d5c040a5dff13503c33e723ec339fb68

### Console Error

```
When writing the command in "Steps to reproduce", this is the output: https://mclo.gs/Z39qlgq
When rejoining immediately after (while "NBT banned"), the output is shorter: https://mclo.gs/JmcMk6C
```

### Bug Description

Having an item with invalid NBT data in your inventory causes an exception on the server-side, which causes the client to be unable to connect.

This issue does **NOT** occur on the latest stable release `ViaVersion-4.10.2`, only the dev branch.

### Steps to Reproduce

1. Run paper server with 1.20.4 with ViaVersion (latest snapshot)
2. Join server on a vanilla 1.20.6 client as an operator
3. Enter the following command into chat: `/give @s barrel{BlockEntityTag:{Items:[{Id:"minecraft:redstone"}]}}`
4. Before running, observe that `Id` is spelled wrong; that should have a lowercase `i` in `id` for this to be a valid item tag.
5. Run the command
6. Observe that the client is disconnected

### Expected Behavior

The client should not be disconnected. They should stay connected and able to play, If the player is using 1.20.4, nothing bad happens, and 1.20.6 clients should have the same experience.

### Additional Server Info

I'm running the server i've tested this on through Velocity, which has [my `ChattORE` working tree](https://github.com/sodiboo/ChattORE/tree/pride) and LuckPerms installed. ChattORE is a plugin that handles chat messages network-wide, DMs, discord bridging, mail, DMs, socialspy, commandspy, helpop, and nicknames. It is the main plugin i am using; the others are dependencies. ChattORE does not affect items or the inventory in any way whatsoever, so this should not be a problem.

ViaVersion is installed on Paper, and the only other plugin on that server is `ChatCancellORE` which just removes chat messages and join/leave messages; these are supposed to be handled by `ChattORE` on the proxy.

I've reproduced this using [this build of ViaVersion](https://modrinth.com/plugin/viaversion/version/5.0.0-SNAPSHOT+409), which is to my knowledge based on the latest commit to the dev branch. I've tested this on the exact same server with Viaversion stable 4.10.2, which does *not* have this issue.

### Checklist

- [X] Via plugins are only running on **EITHER** the backend servers (e.g. Paper) **OR** the proxy (e.g. BungeeCord), **not on both**.
- [X] I have included a ViaVersion dump.
- [X] If applicable, I have included a paste (**not a screenshot**) of the error.
- [X] I have tried the latest build(s) from https://ci.viaversion.com/ and the issue still persists.

**Comment by Barvalg** [comment]
![Platform: git--Paper--497%20%28MC%3A%201.20.4%29](https://img.shields.io/badge/Platform-git--Paper--497%20%28MC%3A%201.20.4%29-3498db?style=flat-square) </br>![ViaVersion (5.0.0-SNAPSHOT): Even with dev](https://img.shields.io/badge/ViaVersion%20(5.0.0--SNAPSHOT)-Even%20with%20dev-brightgreen?style=flat-square) </br>

**Comment by Jo0001** [comment]
Can you please try it with the latest builds?
https://hangar.papermc.io/ViaVersion/ViaVersion/versions?channel=Snapshot
https://hangar.papermc.io/ViaVersion/ViaBackwards/versions?channel=Snapshot
https://hangar.papermc.io/ViaVersion/ViaRewind/versions?channel=Snapshot

**Comment by sodiboo** [comment]
It appears to have been fixed in the latest build `5.0.2-SNAPSHOT+460`. Cannot repro anymore with an identical setup; i tested before and after changing out the ViaVersion plugin on my server.
